cli/lsp/testing/server.rs

```diff
@@ -82,6 +80,7 @@ impl TestServer {
     let _update_join_handle = thread::spawn(move || {
       let runtime = create_basic_runtime();
 
+      #[allow(clippy::await_holding_lock)]
```

Comment:
I think we should address this since it's not a false positive, easiest fix is probably to just use a `tokio::sync::Mutex`